spp_change_request/models/change_request.py

```diff
@@ -177,37 +178,57 @@ def on_submit(self):
 
     def on_validate(self):
         for rec in self:
-            if rec.request_type_ref_id:
-                if rec.state == "pending":
-                    rec.request_type_ref_id._on_validate(rec)
+            # Check if change user is assigned to current user
+            if rec._check_user("Validate"):
+                if rec.request_type_ref_id:
+                    if rec.state == "pending":
+                        rec.request_type_ref_id._on_validate(rec)
+                    else:
+                        raise ValidationError(
+                            _("The request to be validated must be in submitted state.")
+                        )
                 else:
-                    raise ValidationError(
-                        _("The request to be validated must be in submitted state.")
+                    raise UserError(
+                        _("The request details must be properly filled-up.")
                     )
-            else:
-                raise UserError(_("The request details must be properly filled-up."))
 
     def apply(self):
         for rec in self:
-            if rec.state == "validated":
-                rec.request_type_ref_id._apply(rec)
-            else:
-                raise ValidationError(
-                    _(
-                        "The request must be in validated state for changes to be applied."
+            # Check if change user is assigned to current user
+            if rec._check_user("Apply"):
+                if rec.state == "validated":
+                    rec.request_type_ref_id._apply(rec)
+                else:
+                    raise ValidationError(
+                        _(
+                            "The request must be in validated state for changes to be applied."
+                        )
                     )
-                )
 
     def on_reject(self):
         for rec in self:
-            if rec.state in ("draft", "pending"):
-                rec.request_type_ref_id._on_reject(rec)
+            # Check if change user is assigned to current user
+            if rec._check_user("Reject"):
+                if rec.state in ("draft", "pending"):
+                    rec.request_type_ref_id._on_reject(rec)
+                else:
+                    raise UserError(
+                        _(
+                            "The request to be rejected must be in draft or pending validation state."
+                        )
+                    )
+
+    def _check_user(self, process):
+        self.ensure_one()
+        if self.assign_to_id:
+            if self.assign_to_id.id == self.env.user.id:
+                return True
             else:
                 raise UserError(
-                    _(
-                        "The request to be rejected must be in draft or pending validation state."
-                    )
+                    _("You are not allowed to %s this change request") % process
```

Comment:
Use this format or it will be hard to translate:
> _("You are not allowed to %s this change request", process)
https://www.odoo.com/documentation/16.0/developer/howtos/translations.html#variables